src/Ng/CommandHelpers.cs

```diff
@@ -501,6 +551,46 @@ public static Lucene.Net.Store.Directory GetLuceneDirectoryImpl(IDictionary<stri
                     return null;
                 }
 
+                string vaultName;
+                if (!arguments.TryGetValue("-vaultName", out vaultName))
+                {
+                    useKeyVault = false;
+                }
+
+                if (useKeyVault)
```

Comment:
There appears to be a lot of shared keyvault initialization code between here and the `if (useKeyVault)` above...could we make somehow reduce this?